src/Aspire.Hosting.Yarp/YarpNpmResourceExtensions.cs

```diff
@@ -0,0 +1,209 @@
+        var resourceBuilder = builder.AddResource(resource)
+            .WithHttpEndpoint(name: "http", targetPort: 5000)
+            .WithImage("placeholder") // Temporary image, will be replaced by WithDockerfile
+            .WithImageRegistry(null)
+            .WithImageTag("latest")
+            .WithEnvironment("ASPNETCORE_ENVIRONMENT", builder.Environment.EnvironmentName)
+            .WithEntrypoint("dotnet")
+            .WithArgs("/app/yarp.dll")
+            .WithOtlpExporter();
+
+        if (builder.ExecutionContext.IsRunMode)
+        {
+            resourceBuilder.WithEnvironment("YARP_UNSAFE_OLTP_CERT_ACCEPT_ANY_SERVER_CERTIFICATE", "true");
+        }
+
+        // Add the static build options annotation
+        resourceBuilder.WithAnnotation(new NodeStaticBuildOptionsAnnotation(options));
+        
+        // Enable static file serving
+        resourceBuilder.WithStaticFiles();
+        
+        // Add the Dockerfile factory (this will replace the placeholder image)
+        resourceBuilder.WithDockerfile(fullyQualifiedWorkingDirectory, context =>
+        {
+            var yarpResource = (YarpNpmResource)context.Resource;
+            var optionsAnnotation = yarpResource.Annotations.OfType<NodeStaticBuildOptionsAnnotation>().Single();
+            var buildOptions = optionsAnnotation.Options;
+
+            var dockerfile = GenerateDockerfile(buildOptions);
+            return Task.FromResult(dockerfile);
+        });
+
+        return resourceBuilder;
+    }
+
+    /// <summary>
+    /// Configures the package manager to use for the Node.js build.
+    /// </summary>
+    /// <param name="builder">The resource builder for YARP npm resource.</param>
+    /// <param name="packageManager">The package manager (e.g., "npm", "pnpm", "yarn").</param>
+    /// <returns>The <see cref="IResourceBuilder{YarpNpmResource}"/>.</returns>
+    [Experimental("ASPIREHOSTING001", UrlFormat = "https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+    public static IResourceBuilder<YarpNpmResource> WithPackageManager(
+        this IResourceBuilder<YarpNpmResource> builder,
+        string packageManager)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(packageManager);
+
+        var options = GetOrCreateOptions(builder);
+        options.PackageManager = packageManager;
+        return builder;
+    }
+
+    /// <summary>
+    /// Configures the install command for the Node.js build.
+    /// </summary>
+    /// <param name="builder">The resource builder for YARP npm resource.</param>
+    /// <param name="command">The install command (e.g., "install", "ci", "install --frozen-lockfile").</param>
+    /// <returns>The <see cref="IResourceBuilder{YarpNpmResource}"/>.</returns>
+    [Experimental("ASPIREHOSTING001", UrlFormat = "https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+    public static IResourceBuilder<YarpNpmResource> WithInstallCommand(
```

Comment:
Nit: I sort of prefer a single `WithStaticFileOptions` method that takes a callback instead of separate methods for each property in the options.

Reply:
Nested lambdas 😢.

```C#
builder.AddYarpNpmApp("app", "my-react-app").WithStaticFileOptions(o => 
{
 o.Buildcommand = "";
});
```

I dunno....